Week3/homework/step2-2.js

```diff
@@ -3,18 +3,28 @@
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
 
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  for (let number = startIndex; number <= stopIndex; number++) {
```

Comment:
Couple things here:

1. You need not write separate for loops here for creating the numbers array and then iterating it for callback logic. You can combine both these actions in a single for loop like
```
for (let i = startIndex; i <= stopIndex; i++){
  numbers.push(i);
  if(i % 5 === 0 & i % 3 === 0) {
    // and so on
  }
}
```
Having two loops here doesn't impact much in this case since the iteration is very small. However, imagine a situation where you have to loop over 1 or 10 million records. It would be better to run the loop only once, wouldnt' it?
2. Also, I see that you haven't implemented the third requirement which requires to call both callbacks if a number is divisible by both 3 and 5. Could you please add that as well?
Hint: mathematically speaking, a number which is divisible by 3 and 5 will be divisible by 15 as well. Thus, you can also write this logic as simply if (numbers[ii] % 15 === 0) instead of having the && condition. 🙂

Reply:
aah I completely forgot about the third one while doing this 🗡 I hope it is fixed now ^^